File: archive-service/src/main/java/com/wspa/vask/archiveservice/mapper/DocumentMapper.java
package com.wspa.vask.archiveservice.mapper;

import com.wspa.vask.archiveservice.kafka.DocumentMessage;
import com.wspa.vask.archiveservice.document.model.DescriptorImpl;
import com.wspa.vask.archiveservice.document.model.Document;
import com.wspa.vask.archiveservice.document.model.DocumentImpl;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class DocumentMapper {

    public DocumentImpl documentMessageToDocument(DocumentMessage documentMessage) {
        return null;
    }
}

File: back-end/archiveq-parrent/search/src/main/kotlin/com/pl/vkorol/search/mapper/DocumentMapper.kt
package com.pl.vkorol.search.mapper

import com.pl.vkorol.search.model.kafka.DocumentMessage
import com.pl.vkorol.search.model.index.DocumentIndex
import org.springframework.stereotype.Component

@Component
class DocumentMapper {

    fun toDocumentIndex(documentMessage: DocumentMessage): DocumentIndex {
        val documentIndex: DocumentIndex = DocumentIndex(
            documentMessage.documentUuid,
            documentMessage.name,
            "null",
            documentMessage.descriptors
        );
        return documentIndex;
    }
}
